### PR TITLE
sndio init and add support

### PR DESCRIPTION
**Pull request by ajs124**
###### Motivation for this change

I was setting up a distributed audio setup and was evaluating sndio as one of the options. While I did not end up using it, it seems like a really handy tool.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Todo

Add a module. Add tests.

---

### CODE REVIEW

**Comment by ajs124** [comment]
##### mpd maintainers
@astsmtl @fuuzetsu @ehmry @fpletz: Since the mpd package seems to have a "turn on all the features" philosophy, I just followed that and turned on sndio by default as well. I can obviously change that, if you disagree with that.

##### mpv maintainers
@AndersonTorres @fuuzetsu @fpletz: Unlike mpd, mpv seems more conservative in what is on by default, so I choose to leave sndioSupport false, by default.

##### cmus maintainer
@oxij: You had this as a TODO, from when you rewrote this at the end of 2015. It's here now! :D

File: pkgs/applications/audio/pulseaudio-module-sndio/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    owner = "t6";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256:1xngzqhg01df9fgadv10nmjxpzsykycbxrs241ilpr98x1vf1kn5";
+  };
+
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ sndio pulseaudio libtool ];
+  makeFlags = [
+    "DESTDIR=$$out"
```
Comment (symphorien) [inline pkgs/applications/audio/pulseaudio-module-sndio/default.nix]:
```suggestion
    "DESTDIR=${placeholder "out"}"
```
This is probably less hacky.

Comment (ajs124) [reply pkgs/applications/audio/pulseaudio-module-sndio/default.nix]: Done. There's a few places in nixpkgs which do it exactly like I did, but the majority follows your suggestion. Maybe those should be refactored.

**Comment by oxij** [comment]
+let
+  pulsecore = runCommand "pulseaudio-sources" {} ''
+    tar -xf ${pulseaudio.src}
+    mv pulseaudio*/src/pulsecore $out
+  '';
+in
+  stdenv.mkDerivation rec {
+  pname = "pulseaudio-module-sndio";

Bad indent. I would do

+in stdenv.mkDerivation rec {

Also

-    description = "A small audio and MIDI framework, originally part of the OpenBSD project.";
+    description = "A small sound server and an audio and MIDI framework, originally part of the OpenBSD project.";

In general, it would be nice to split this into sndiod, libsndio and tools (aucat and etc).

In which case I would vote for enabling libsndio support in all packages by default, as it is tiny (73Kb, which is smaller than even libcardiacarrest at 87Kb, because libpulse has such a huge number of symbols that just stubbing them out takes 87Kb) and nixpkgs has libpulseaudio with its megabites of closure size enabled by default anyway.

If you are feeling lazy, then please at least make `libsndio` an alias to `sndio` and refer as `libsndio` in other packages, I'll split `sndio` into pieces when I have time.

**Comment by ajs124** [comment]
I split it into "bin" and "dev" and assigned libsndio = sndio.dev, does that work for you?

**Comment by oxij** [comment]
Please apply the following, else the contents of the outputs is incorrect.

```diff
diff --git a/pkgs/servers/sndio/default.nix b/pkgs/servers/sndio/default.nix
index aaaaaaaaaaa..bbbbbbbbbbb 100644
--- a/pkgs/servers/sndio/default.nix
+++ b/pkgs/servers/sndio/default.nix
@@ -11,11 +11,17 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ alsaLib ];
 
-  outputs = [ "bin" "dev" "out" ];
+  outputs = [ "out" "dev" "man" ];
   setOutputFlags = false;
 
+  postInstall = ''
+    moveToOutput lib $dev
+    moveToOutput include $dev
+    moveToOutput share/man $man
+  '';
+
   meta = with stdenv.lib; {
-    description = "A small sound server and an audio and MIDI framework, originally part of the OpenBSD project.";
+    description = "A small sound server and an audio and MIDI framework, originally part of the OpenBSD project";
     homepage    = http://www.sndio.org/;
     license     = licenses.isc;
     maintainers = with maintainers; [ ajs124 ];
```

**Comment by ajs124** [comment]
sndio was init'ed by #61742, but didn't add support to any applications. The version in there is also quite out of date.

**Pull request closed without merging**